app/javascript/components/supplemental_application/sections/Lease.js

```diff
@@ -162,34 +168,43 @@ const Lease = ({ form, values, store }) => {
 
   const unitIsUsedWithPrefElsewhere = (unit, pref) =>
     unit.application_id &&
-    unit.application_id !== application.id &&
+    unit.application_id !== state.application.id &&
     unit.preference_used_name === pref
 
   const numUnitsUsingPref = (pref) =>
-    units.filter(
+    state.units.filter(
       (unit) =>
         (unit.id === selectedUnit && isSelected(pref)) || unitIsUsedWithPrefElsewhere(unit, pref)
     ).length
 
   const remainingSetAsidesForPref = (pref) =>
-    totalSetAsidesForPref(listing, pref) - numUnitsUsingPref(pref)
+    totalSetAsidesForPref(state.listing, pref) - numUnitsUsingPref(pref)
 
   // Selected unit does not count in available a11y units.
   const remainingPriorityUnits = availableUnits.filter(
     (unit) => unit.id !== selectedUnit && unit.priority_type?.match(/Mobility|Hearing|Vision/)
   )
 
+  const handleCancelLeaseClick = (form) => {
+    leaseCanceled(dispatch, state.application)
+
+    // Reset lease and assistances on the form
+    form.change('lease', state.application.lease)
```

Comment:
I couldn't figure out a better way to integrate final form with the app context. It would be nice if all of this form data lived in the context as well, but it's not really feasible with the final form version we're using now.